File: implementations/python-engram-mesh/adapter.py
#!/usr/bin/env python3
"""Dependency-free Engram Mesh 0.3 prototype adapter."""
import json, shutil, sys
from pathlib import Path
ROOT = Path(__file__).resolve().parents[2]
sys.path.insert(0, str(ROOT / "scripts"))
from validate_engram_mesh import validate
def main():
    operation, request_path = sys.argv[1:]
    request = json.loads(Path(request_path).read_text()); source = Path(request["fixture"])
    mesh = source / "engram-mesh.json"; validate(mesh)
    artifacts = []
    if operation in {"produce", "round-trip"}:
        out = Path(request["artifact_directory"]) / "engram-mesh.json"; shutil.copyfile(mesh, out); artifacts = [{"path":"engram-mesh.json", "media_type":"application/json"}]
    print(json.dumps({"protocol_version":"1.0","case_id":request["case_id"],"outcome":"completed","observed":{"status":"success","mesh_id":json.loads(mesh.read_text())["mesh_id"]},"diagnostics":[],"artifacts":artifacts}))
if __name__ == "__main__": main()

File: implementations/node-engram-mesh/adapter.js
#!/usr/bin/env node
const fs = require('fs'); const path = require('path');
const request = JSON.parse(fs.readFileSync(process.argv[3], 'utf8'));
const mesh = path.join(request.fixture, 'engram-mesh.json'); const value = JSON.parse(fs.readFileSync(mesh, 'utf8'));
if (value.format !== 'engram-mesh' || value.version !== '0.3') throw new Error('unsupported Engram Mesh document');
for (const key of ['sources', 'nodes', 'bindings', 'relationships']) if (!Array.isArray(value[key])) throw new Error(`missing ${key}`);
const nodeIds = new Set(value.nodes.map(node => node.id)), sourceIds = new Set(value.sources.map(source => source.id));
for (const binding of value.bindings) if (!nodeIds.has(binding.node) || !sourceIds.has(binding.source)) throw new Error('unresolved source binding');
for (const relationship of value.relationships) if (!nodeIds.has(relationship.from) || !nodeIds.has(relationship.to)) throw new Error('unresolved relationship');
let artifacts = [];
if (['produce', 'round-trip'].includes(process.argv[2])) { fs.copyFileSync(mesh, path.join(request.artifact_directory, 'engram-mesh.json')); artifacts = [{path: 'engram-mesh.json', media_type: 'application/json'}]; }
process.stdout.write(JSON.stringify({protocol_version:'1.0', case_id:request.case_id, outcome:'completed', observed:{status:'success',mesh_id:value.mesh_id}, diagnostics:[], artifacts})+'\n');
